src/Features/CSharp/Portable/UseObjectInitializer/CSharpUseObjectInitializerDiagnosticAnalyzer.cs

```diff
@@ -21,6 +21,11 @@ internal class CSharpUseObjectInitializerDiagnosticAnalyzer :
     {
         protected override bool FadeOutOperatorToken => true;
 
+        // object initializers are only available in C# 3.0 and above.  Don't offer this refactoring
+        // in projects targeting a lesser version.
+        protected override bool AreObjectInitializersSupported(SyntaxNodeAnalysisContext context) =>
+            ((CSharpParseOptions) context.Node.SyntaxTree.Options).LanguageVersion >= LanguageVersion.CSharp3;
```

Comment:
I'd also prefer the comments to be closer to the logic (for here and in the VB one too). That reads as more a generic doc comment.